tests/test_plugin.py

```diff
@@ -46,24 +46,57 @@ def test_save_raises_entity_creation_error_on_creation_error(
 
     def mock_driver_error(*args, **kwargs):
         raise BigchaindbException()
-    monkeypatch.setattr(plugin.driver.transactions, 'create',
+    monkeypatch.setattr(plugin.driver.transactions, 'prepare',
                         mock_driver_error)
 
     with raises(EntityCreationError):
         plugin.save(manifestation_model_json, user=alice_keypair)
 
 
-def test_save_raises_persistence_error_on_error(monkeypatch, plugin,
-                                                manifestation_model_json,
-                                                alice_keypair):
-    from coalaip.exceptions import PersistenceError
+def test_save_raises_entity_creation_error_on_missing_key(monkeypatch, plugin,
+                                                          manifestation_model_json,
+                                                          alice_keypair):
+    from bigchaindb_driver.exceptions import MissingSigningKeyError
+    from coalaip.exceptions import EntityCreationError
 
     def mock_driver_error(*args, **kwargs):
-        raise Exception()
-    monkeypatch.setattr(plugin.driver.transactions, 'create',
+        raise MissingSigningKeyError()
+    monkeypatch.setattr(plugin.driver.transactions, 'fulfill',
                         mock_driver_error)
 
-    with raises(PersistenceError):
+    with raises(EntityCreationError):
+        plugin.save(manifestation_model_json, user=alice_keypair)
+
+
+def test_save_raises_entity_creation_error_on_transport_error(monkeypatch,
```

Comment:
@sohkai 

- `test_save_raises_entity_creation_error_on_transport_error` and
- `test_save_raises_entity_creation_error_on_connection_error`

Is this how you intended it?